packages/cli/src/lib/autorun/command-object.spec.ts

```diff
@@ -51,13 +55,20 @@ const cli = (args: string[]) =>
     commands: [yargsAutorunCommandObject()],
   });
 
+let logs = [];
+
 describe('autorun-command-object', () => {
   beforeEach(async () => {
     vi.clearAllMocks();
     cleanFolderPutGitKeep();
+    mockConsole((...args: unknown[]) => {
+      logs.push(...args);
+    });
   });
   afterEach(() => {
     cleanFolderPutGitKeep();
+    logs = [];
+    unmockConsole();
   });
```

Comment:
Seeing how the console mocking is being used, it feels unnecessarily boilerplatey. E.g. the caller still has to be manage the `logs` array themselves, despite the logic always being the same.

If we were to just go with Vitest's built-in spies, it would actually be simpler:

```suggestion
describe('autorun-command-object', () => {
  let logSpy: SpyInstance;

  beforeEach(async () => {
    vi.clearAllMocks();
    cleanFolderPutGitKeep();
    logSpy = vi.spyOn(console, 'log');
  });

  afterEach(() => {
    cleanFolderPutGitKeep();
    logSpy.mockRestore();
  });
```

The only change to the assertions is:
- `expect(logs).toContain('...')` becomes `expect(console.log).toHaveBeenCalledWith('...')`,
- `expect(logs.length).toBe(2)` becomes `expect(console.log).toHaveBeenCalledTimes(0)`.

Reply:
> If we were to just go with Vitest's built-in spies, it would actually be simpler:

I was not able to find that in the docs. Thanks!

Reply:
Actually i cant reproduce this... still see the logs in the console

Reply:
I can't help you with that, unfortunately. Tests keep failing on my machine since the big testing refactor in #134 :disappointed: 

![image](https://github.com/flowup/quality-metrics-cli/assets/34691111/4d0b1980-cd66-4772-9047-8fe9af24861d)

Reply:
will address the flakyness in the next PR